### PR TITLE
Ballchasy

### PR DESCRIPTION
**Pull request by BenjaminHunt**
# bcManager
- Enables getting and setting of statsManager role. Enables stats managers to execute necessary maintenance commands
- Includes match day param to `bcr` command
- Adds `manualMatchUpdate <ballchasing link>` to have a scoreline updated based on what exists at the provided group linked
- Adds `manualMatchReport <match day> <team a> <team a wins> <team b> <team b wins>` to manually report a match apart from a ballchasing group
- Adds `reportForfeit <match day> <team a> <team b>` to update match for forfeited match games with Embed/Reaction menu
- Adds support for RSC web app tokens
- Updates set token command name for ballchasing
- Updates code to use accounts from RSC web service

### CODE REVIEW

File: bcManager/bcManager.py [hunk 1]
```diff
@@ -773,9 +1404,45 @@ def get_replay_teams_and_players(self, replay):
...
+        if 'group/' in code_or_link:
+            code_or_link = code_or_link.split('group/')[-1]
+        
+        return code_or_link  # returns code
+
+    async def get_player_accounts(self, player: discord.Member, platforms=[]):
+        url =  f"{RSC_WEB_APP}/api/member/{player.id}/accounts"
+        rsc_app_token: str = await self._get_rsc_app_token(player.guild)
+        headers = {"X-Api-Key": rsc_app_token}
+        data = requests.get(url, headers=headers).json()
```
Comment (adam-stover) [inline bcManager/bcManager.py:1429]: would we prefer `aiohttp` or similar async networking library so we can do this w/o blocking?

Comment (BenjaminHunt) [reply bcManager/bcManager.py:1429]: We already use to_thread in a bunch of places, I can update it to be consistent with that.

File: bcManager/bcManager.py [hunk 2]
```diff
@@ -217,16 +435,117 @@ async def bcgroup(self, ctx):
...
+
+    async def process_ff_reacts(self, reaction: discord.Reaction, user: discord.User, added: bool):
+        try:
+            message = reaction.message
+            member = message.author
+            guild = message.guild
+            channel = message.channel
+            ff_processing_data = self.ffp[guild][message]
+            if user.id is not ff_processing_data['reporter'].id:
+                await channel.send("WHAT")
```
Comment (adam-stover) [inline bcManager/bcManager.py:528]: what's this lol

File: bcManager/bcManager.py [hunk 3]
```diff
@@ -307,21 +626,294 @@ async def process_match_bcreport(self, ctx, match, tier_md_group_code: str=None,
...
+            react_hex = hex(react_hex_code+i)
+            react = self.get_select_reaction(int(react_hex, base=16))
+            ff_able_reacts.append(react)
+            home_goals, away_goals = self.get_home_away_goals(match, replay)
+            winner_emoji = home_emoji if home_goals > away_goals else away_emoji
+            summary = f"{react} **G{gi}:** {match['home']} {home_goals} - {away_goals} {match['away']}"
+            if winner_emoji:
+                summary += f" {winner_emoji}"
+
+            game_summaries.append({
```
Comment (adam-stover) [inline bcManager/bcManager.py:685]: can you include an example summary string as a comment in this section so we know what it should look like? would help debug if anything goes wrong or if it gets more complex in future

File: bcManager/bcManager.py [hunk 4]
```diff
@@ -880,17 +1554,76 @@ async def get_stats_updates_channel(self, guild: discord.Guild):
...
+                    and
+                    player.get("id", {}).get('id', None) == platform_id
+                )
+                if account_match:
+                    return player
+        return {}
+    
+    def get_latest_player_data_by_platform_name(self, guild, platform, plat_name):        
+        bapi : ballchasing.Api = self.ballchasing_api[guild]
+        rsc_steam_id = "76561199096013422"
```
Comment (adam-stover) [inline bcManager/bcManager.py:1605]: what's this magic number, should it be in code vs as an env variable or..?

**Review by adam-stover — APPROVED** [review]
More comments above the new functions would be really helpful. But the code looks effective, very nice stuff Null!

File: bcManager/bcManager.py (hunk 2 above)
Comment (BenjaminHunt) [reply bcManager/bcManager.py:528]: LOL I was troubleshooting some logic. Forgot to remove it. I'll get that taken care of.

File: bcManager/bcManager.py (hunk 4 above)
Comment (BenjaminHunt) [reply bcManager/bcManager.py:1605]: That's my personal discord id, I need to update that out as well.

File: bcManager/bcManager.py (hunk 3 above)
Comment (BenjaminHunt) [reply bcManager/bcManager.py:685]: yes

File: bcManager/bcManager.py [hunk 5]
```diff
@@ -184,27 +241,188 @@ async def reportMatches(self, ctx, match_day: int=None):
...
+            "status": "active",
+            "timeout": datetime.now() + timedelta(seconds=15),
+            "ctx": ctx
+        }
+
+        self.ffp.setdefault(ctx.guild, {}).setdefault(message, msg_val)
+        
+        await self.assign_ff_reactions(message, deep_match_report)
+
+    @commands.command(aliases=['mmr'])
```
[the remark this answers is not in the document]
Comment (BenjaminHunt) [reply bcManager/bcManager.py:276]: I'll take it out, but I could see this being a pretty common command, so I think we should think of and implement a replacement when we can. Perhaps some input from stats committee will help.

File: bcManager/bcManager.py (hunk 4 above)
Comment (BenjaminHunt) [reply bcManager/bcManager.py:1605]: lol alright the above is actually incorrect. I believe that's exactly what its named: That is the Steam ID for the RSC account associated with the league.

I'm gonna leave it in for now. We should probably have it be a configurable value in the future.

**Comment by BenjaminHunt** [comment]
I'm the best

**Comment by BenjaminHunt** [comment]
I'm not the best, I missed a push lol

**Pull request merged**